### PR TITLE
MattT/APPEALS-10821: Alteration of VHA Benefit Type Selection for Non-VHA Users

### PR DESCRIPTION
**Pull request by ThorntonMatthew**
Resolves [APPEALS-10821](https://vajira.max.gov/browse/APPEALS-10821)

[Diff Link](https://github.com/department-of-veterans-affairs/caseflow/compare/feature/APPEALS-8982..MattT/APPEALS-10821) | [Designs Link](https://bipuiux.invisionapp.com/console/share/QTCKRDXAHEN/953715692/play)

### Description
On the Review page for claim review intakes (HLRs and SCs) the "Veterans Health Administration" benefit type radio field option will be disabled if the user is not a member of the VHA business line. In addition, if the non-VHA user hovers over the VHA benefit type option they will be presented with an information tooltip.

### Acceptance Criteria
  - [x] A non-VHA user (user that is not a member of the Veterans Health Administration business line organization in Caseflow) will be able to view the following benefit types when intaking Higher-Level Reviews and Supplemental Claims: 
    - [x] Compensation 
    - [x] Pension & Survivor’s Benefits 
    - [x] Fiduciary  
    - [x] Insurance 
    - [x] Education 
    - [x] Veterans Readiness and Employment 
    - [x] Loan Guaranty  
    - [x] National Cemetery Administration  
    - [x] Veteran Health Administration
  - [x] Order of benefit types is changed to place Veterans Health Administration after National Cemetery Organization when user is on the Review page  
  - [ ] The following occurs when a non-VHA user attempts to select Veterans Health Administration as a benefit type: 
    - [x] Option will not be selectable 
    - [x] Option will be grayed out in its entirety  
    - [x] When hovered over, a message will appear with the following informational text:  
      - [x] Only VHA case team members can establish Higher Level Reviews and Supplemental Claims with VHA issues. If you have a VHA claim, please return the packet to VHA queue in the Centralized Mail Portal or send downloaded documents to VHABENEFITAPPEALS@va.gov.

### Testing Plan

1. Open the Rails console and add an Intake user to the VHA business line:
```ruby
BusinessLine.find_by(url: "vha").add_user(User.find_by_css_id("BVAISHAW"))
```

2. Log into Caseflow as this user. Navigate to Caseflow Intake ([Local link](http://localhost:3000/intake))

3. Start a Higher-Level Review intake.

4. Ensure in the "What is the Benefit Type?" section that the "Veterans Health Administration" radio field is enabled and selectable.

5. Cancel the intake, and repeat steps 3 and 4 for a Supplemental Claim intake.

6. Log into Caseflow as an Intake user that is **not** a member of the VHA business line (ex: `BVADWISE` should work).

7. Start another Higher-Level Review intake.

8. This time, the "Veterans Health Administration" option should be **disabled**. Ensure that it is not selectable.

9. Hover your cursor over the VHA benefit type option. A tooltip should appear. [Click here](https://bipuiux.invisionapp.com/console/share/QTCKRDXAHEN/953715692/play) for the associated design mockups to compare its appearance with what was specified.

10. Cancel this intake and repeat steps 7-9 for a Supplemental Claim.

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 USER|BEFORE|AFTER
 ---|---|---
 VHA|before|vha-after
 Non-VHA|before|vha-after

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [x] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [x] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [x] Write a separate story (within the same file) for each discrete variation of the component

storybook

### CODE REVIEW

File: client/app/styles/_commons.scss [hunk 1]
```diff
@@ -954,7 +954,6 @@ fieldset {
 
   .disabled {
     color: $color-gray-light;
-    font-style: italic;
```
Comment (ThorntonMatthew) [inline client/app/styles/_commons.scss:957]: See this [message](https://bah-ssp.slack.com/archives/C03N1N59UH0/p1668096292899779?thread_ts=1668085034.809639&cid=C03N1N59UH0) for context on why this was removed.

File: client/app/components/Tooltip.jsx [hunk 2]
```diff
@@ -21,7 +22,7 @@ const Tooltip = (props) => {
       backgroundColor: COLORS.GREY_DARK,
       fontWeight: 'normal',
       padding: '0.5rem 1rem',
-      textAlign: 'center'
+      textAlign: 'justify'
```
Comment (ThorntonMatthew) [inline client/app/components/Tooltip.jsx:25]: See [this message](https://bah-ssp.slack.com/archives/C03N1N59UH0/p1668086121193569?thread_ts=1668085034.809639&cid=C03N1N59UH0) for the "justification" behind this change.

File: spec/feature/intake/review_page_spec.rb [hunk 3]
```diff
@@ -439,6 +439,79 @@ def select_claimant(index = 0)
...
+      let(:vha_business_line) { create(:business_line, name: benefit_type_label, url: "vha") }
+      let(:current_user) { create(:user, roles: ["Admin Intake"]) }
+
+      before do
+        vha_business_line.add_user(current_user)
+        User.authenticate!(user: current_user)
+      end
+
+      it "VHA benefit type radio option is enabled" do
+        expect(page).to have_field benefit_type_label, disabled: false, visible: false
```
Comment (ThorntonMatthew) [inline spec/feature/intake/review_page_spec.rb:457]: The tests for HLRs and SCs will fail (two failures) on this line until APPEALS-10818 is merged in.

File: app/controllers/intakes_controller.rb [hunk 4]
```diff
@@ -97,6 +97,7 @@ def index_props
     {
       userDisplayName: current_user.display_name,
       userCanIntakeAppeals: current_user.can_intake_appeals?,
+      userIsVhaEmployee: false, # Will be replaced by 10818
```
Comment (ThorntonMatthew) [inline app/controllers/intakes_controller.rb:100]: This is where APPEALS-10818 will be joined with this story.

File: spec/feature/intake/higher_level_review_spec.rb [hunk 5]
```diff
@@ -52,10 +52,12 @@
   before do
     FeatureToggle.enable!(:filed_by_va_gov_hlr)
     FeatureToggle.enable!(:updated_intake_forms)
+    FeatureToggle.enable!(:use_ama_activation_date)
```
Comment (ThorntonMatthew) [inline spec/feature/intake/higher_level_review_spec.rb:55]: Fixes a feature test that is failing in master.

File: spec/feature/reader/reader_spec.rb [hunk 6]
```diff
@@ -695,7 +695,11 @@ def placing_annotation_icon_position
         expect(page).to have_css(".page")
 
         # Click on the second to last comment icon (last comment icon is off screen)
-        all(".commentIcon-container", wait: 3, count: documents[0].annotations.size)[annotations.size - 4].click
+        all(
```
Comment (ThorntonMatthew) [inline spec/feature/reader/reader_spec.rb:698]: Fixes a feature test that is failing in master.

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -746,12 +746,12 @@ class AnotherFakeTask < Dispatch::Task; end
 
   describe "vha_employee?" do
     let(:user) { create(:user) }
-    let(:org) { BusinessLine.create!(name: "Veterans Health Administration", url: "vha" ) }
+    let(:org) { BusinessLine.create!(name: "Veterans Health Administration", url: "vha") }
```
Comment (ThorntonMatthew) [inline spec/models/user_spec.rb:749]: CC fix

File: spec/models/user_spec.rb [hunk 8]
```diff
@@ -746,12 +746,12 @@ class AnotherFakeTask < Dispatch::Task; end
...
 
     subject { user.vha_employee? }
 
     context "when current user is a member of VHA org" do
-      before { org.add_user(user)}
+      before { org.add_user(user) }
```
Comment (ThorntonMatthew) [inline spec/models/user_spec.rb:754]: CC fix

File: client/app/intake/reducers/userInformation.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    },
+  });
+};
+
+export const mapDataToUserInformation = (data = { userInformation: {} }) =>
+  updateFromServerUserInformation(
+    {
+      userDisplayName: null,
+      userCanIntakeAppeals: false,
+      userIsVhaEmployee: false,
```
Comment (ajspotts) [inline client/app/intake/reducers/userInformation.js:25]: Mind elaborating a little on why it isn't necessary to include `unreadMessages` here?

Comment (ThorntonMatthew) [reply client/app/intake/reducers/userInformation.js:25]: I believe it would be best to include it here. I've just pushed up a commit that adds it.

**Review by ajspotts — APPROVED** [review]
Awesome changes lgtm!

**Pull request merged**